runtime/src/account_storage.rs

```diff
@@ -45,23 +46,31 @@ impl AccountStorage {
         self.map.iter().map(|iter_item| *iter_item.key()).collect()
     }
 
-    pub(crate) fn extend(&mut self, source: AccountStorageMap) {
-        self.map.extend(source.into_iter())
+    /// initialize the storage map to 'all_storages'
+    pub(crate) fn initialize(&mut self, all_storages: AccountStorageMap) {
+        assert!(self.map.is_empty());
+        self.map.extend(all_storages.into_iter())
```

Comment:
Probably better to just replace the memory if we assert it's empty:
```suggestion
        self.map = all_storages;
```